### PR TITLE
feature: add ngx_http_lua_ffi_worker_pids to get all workers pid map

### PR DESCRIPTION
**Pull request by attenuation**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

File: t/122-worker.t [hunk 1]
```diff
@@ -79,3 +79,34 @@ worker pid: \d+
...
+            ngx.say("worker pid is wrong.")
+        ';
+    }
+--- request
+GET /lua
+--- response_body_like
+worker pid: \d+
+worker pid is correct\.
+--- no_error_log
+[error]
```
Comment (zhuizhuhaomeng) [inline t/122-worker.t:112]: need a newline at the end of the file

File: src/ngx_http_lua_worker.c [hunk 2]
```diff
@@ -20,6 +21,36 @@ ngx_http_lua_ffi_worker_pid(void)
 }
 
 
+int
+ngx_http_lua_ffi_worker_pids(int *pids, size_t *pidslen)
```
Comment (zhuizhuhaomeng) [inline src/ngx_http_lua_worker.c:25]:
```suggestion
ngx_http_lua_ffi_worker_pids(int *pids, size_t *pids_len)
```

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: README.markdown [hunk 3]
```diff
@@ -9025,6 +9026,17 @@ This API was first introduced in the `0.9.5` release.
...
 [Back to TOC](#nginx-api-for-lua)
 
+ngx.worker.pids
+--------------
+
+**syntax:** *pids = ngx.worker.pids()*
+
+**context:** *set_by_lua&#42;, rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;, header_filter_by_lua&#42;, body_filter_by_lua&#42;, log_by_lua&#42;, ngx.timer.&#42;, init_by_lua&#42;, init_worker_by_lua&#42;, exit_worker_by_lua&#42;*
+
+This function returns a Lua table for all Nginx worker process ID (PID). Nginx uses channel to send the current worker PID to another worker in the worker process start or restart. So this API can get all worker PID at any time. `init_by_lua` and `init_worker_by_lua` maybe can not get all worker PID because in these phases other worker isn't started.
```
Comment (zhuizhuhaomeng) [inline README.markdown:9036]: can we just disable the API in init_by_lua and init_worker_by_lua

Comment (zhuizhuhaomeng) [inline README.markdown:9036]: I would like to see more explanation why you need this API.
What are the application scenarios?

Please add an example below.

**Review by zhuizhuhaomeng — CHANGES_REQUESTED** [review]

File: README.markdown (hunk 3 above)
Comment (attenuation) [reply README.markdown:9036]: In one worker, we know all worker pid, so we can send a signal or other rpc call to other worker.

Comment (attenuation) [reply README.markdown]: done

**Comment by zhuizhuhaomeng** [comment]
The ci is still failing.
Please modify the .travis.yml, change the lua-resty-core to your repo and the related branch

File: README.markdown (hunk 3 above)
Comment (zhuizhuhaomeng) [reply README.markdown]: https://github.com/openresty/lua-nginx-module#data-sharing-within-an-nginx-worker

You need to add examples to the doc like the above link.

Comment (zhuizhuhaomeng) [reply README.markdown]: Please also add the application scenarios that could inspire people to the doc.

**Comment by zhuizhuhaomeng** [comment]
CI is still failed. 
See https://app.travis-ci.com/github/openresty/lua-nginx-module/jobs/580097060

https://github.com/openresty/lua-nginx-module#data-sharing-within-an-nginx-worker
You need to add examples to the doc like the above link.

**Comment by attenuation** [comment]
@zhuizhuhaomeng I think this is a worker API, you can use this API to know all Nginx worker PID, just an API, not a big function impl, It is not in one level with a big module(eg: ngx_http_lua_shdict), in order to this small API to write a long doc is dirty? And this doc will include some API that is not included by Openresty.

**Comment by attenuation** [comment]
It seems that ci failure is not related to my change.

**Comment by zhuizhuhaomeng** [comment]
merge with the following patch
```
diff --git a/t/062-count.t b/t/062-count.t
index d524da47..ad464ba2 100644
--- a/t/062-count.t
+++ b/t/062-count.t
@@ -438,7 +438,7 @@ thread: 3
 --- request
 GET /test
 --- response_body
-worker: 4
+worker: 5
 --- no_error_log
 [error]
```

**Pull request closed without merging**